sktime/forecasting/model_selection/_split.py

```diff
@@ -412,41 +415,216 @@ def __init__(
+            for idx in y_index_inst:
+                y_inst = y_index_df.loc[idx]
+                y_inst = y_inst.reset_index(drop=True).set_index(y_inst.columns[0])
+                y_inst_index = y_inst.index
+                train_test_res[idx] = list(self._split(y_inst_index))
+                for i, tt in enumerate(train_test_res[idx]):
+                    train_iloc[idx][i] = tt[0]
+                    test_iloc[idx][i] = tt[1]
+                for i, train in train_iloc[idx].items():
+                    train_iloc[idx][i] = y_inst["__index"].iloc[train].values
+                for i, test in test_iloc[idx].items():
+                    test_iloc[idx][i] = y_inst["__index"].iloc[test].values
+
+            train_multi = dict()
+            test_multi = dict()
+            for idx in y_index_inst:
+                for i in train_iloc[idx].keys():
+                    if i not in train_multi.keys():
+                        train_multi[i] = train_iloc[idx][i]
+                        test_multi[i] = test_iloc[idx][i]
+                    else:
+                        train_multi[i] = np.concatenate(
+                            (train_iloc[idx][i], train_multi[i])
+                        )
+                        test_multi[i] = np.concatenate(
+                            (test_iloc[idx][i], test_multi[i])
+                        )
+            for i in train_multi.keys():
+                train_multi[i] = np.sort(train_multi[i])
+                test_multi[i] = np.sort(test_multi[i])
+
+            for i in train_multi.keys():
+                train = train_multi[i]
+                test = test_multi[i]
+                yield train[train >= 0], test[test >= 0]
 
     def _split(self, y: pd.Index) -> SPLIT_GENERATOR_TYPE:
-        """Split `y` into training and test windows.
+        """Get loc references to train/test slits of `y`.
 
         Parameters
         ----------
-        y : pd.Index
-            Index of time series to split
+        y : pd.Index or time series in sktime compatible time series format
+            Time series to split, or index of time series to split
 
         Yields
         ------
-        training_window : np.ndarray
-            Training window indices
-        test_window : np.ndarray
-            Test window indices
+        train : 1D np.ndarray of dtype int
+            Training window indices, iloc references to training indices in y
+        test : 1D np.ndarray of dtype int
+            Test window indices, iloc references to test indices in y
         """
         raise NotImplementedError("abstract method")
 
+    def split_loc(self, y: ACCEPTED_Y_TYPES) -> SPLIT_GENERATOR_TYPE:
+        """Split `y` into training and test windows.
+
+        Parameters
+        ----------
+        y : pd.Index or time series in sktime compatible time series format (any)
+            Time series to split, or index of time series to split
+
+        Yields
+        ------
+        train : pd.Index
```

Comment:
I think it should be
```
1D np.ndarray of dtype int
```
same as in `_split`

Reply:
not in `_split_loc`, this is a new method which returns an index

Reply:
iloc references are always integer
loc references are typically index

Reply:
`split_loc` calls `self.split` which in turn returns `1D np.ndarray of dtype int`. docstrings seem to be inconsistent.

Reply:
> docstrings seem to be inconsistent.

I don't think so? Call `split_loc` an you should see it returns an index.
That this is so is also tested in `test_split_loc`.

Reply:
Sure, makes sense. But, please, find `assert isinstance(window, np.ndarray)` in the tests which clearly states that the return type of `.split` is `np.ndarray`. This resolves to `False`: `isinstance(pd.Index([], dtype=int), np.ndarray)`. Hence, there is inconsistency in both docstrings and in type suggestions (see `SPLIT_GENERATOR_TYPE`).

Reply:
ok, I´m getting confused here.
Can you kindly be precise, and tell me which function precisely, in your opinion, has an incorrect docstring, and which part of the docstring is incorrect, and what it should be instead?

Reply:
Sorry, I got confused myself. Docstrings are ok. Type suggestion is wrong. See another comment to `split_loc` method.